ucs_chassis_connection.py

```diff
@@ -0,0 +1,154 @@
+    - Cisco recommends including information about where and when to use the policy.
+    - Enter up to 256 characters.
+    - "You can use any characters or spaces except the following:"
+    - "` (accent mark), \ (backslash), ^ (carat), \" (double quote), = (equal sign), > (greater than), < (less than), or ' (single quote)."
+    aliases: [ descr ]
+  sioc_connectivity:
+    description:
+    - The server SIOC connectivity that should be used for UCS S-Series.
+	- "single-server-single-sioc - The data path is configured through one SIOC when the chassis has single server and single SIOC or dual server and dual SIOCs."
+	- "single-server-dual-sioc - When enabled, you can configure the data path through both the primary and auxiliary SIOCs when the chassis has single server and dual SIOCs."
+    default: 'single-server-single-sioc'
+requirements:
+- ucsmsdk
+author:
+- Olli Walsdorf (@owalsdor)
+- CiscoUcs (@CiscoUcs)
+version_added: '2.5'
+'''
+
+EXAMPLES = r'''
+- name: Add Chassis Connection Policy
+  ucs_chassis_connection:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    name: s3260_single
+    sioc_connectivity: 'single-server-single-sioc'
+- name: Remove Chassis Firmware Package
+  ucs_chassis_fw:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    name: s3260_single
+    state: absent
+'''
+
+RETURN = r'''
+#
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.remote_management.ucs import UCSModule, ucs_argument_spec
+
+
+def main():
+	argument_spec = ucs_argument_spec
+	argument_spec.update(
+		name=dict(type='str', required=True),
+		sioc_connectivity=dict(type='str', default='single-server-single-sioc', choices=['single-server-single-sioc', 'single-server-dual-sioc']),
+		state=dict(type='str', default='present', choices=['present', 'absent']),
+	)
+
+	module = AnsibleModule(
+		argument_spec,
+		supports_check_mode=True,
+		)
+		
+	ucs = UCSModule(module)
+
+	err = False
+
+	# UCSModule creation above verifies ucsmsdk is present and exits on failure, so additional imports are done below.
+	from ucsmsdk.mometa.equipment.EquipmentComputeConnPolicy import EquipmentComputeConnPolicy
+
+	changed = False
+	try:
+		mo_exists = False
+		props_match = False
+		# dn is org-root//compute-conn-policy-<name> for chassis connection policy
+		dn_base = 'org-root'
+		dn = dn_base + '/compute-conn-policy-' + module.params['name']
+
+		mo = ucs.login_handle.query_dn(dn)
+		if mo:
+			mo_exists = True
+
+		if module.params['state'] == 'absent':
+			# mo must exist but all properties do not have to match
+			if mo_exists:
+				if not module.check_mode:
+					ucs.login_handle.remove_mo(mo)
+					ucs.login_handle.commit()
+				changed = True
+		else:
+			if mo_exists:
+				# check top-level mo props
+				kwargs = dict(name=module.params['name'])
+				kwargs['serverSiocConnectivity'] = module.params['sioc_connectivity']

```

Comment:
Based on the param name in the python class, I'd expect the keyword arg to be 'server_sioc_connectivity'.  Also, since the name is part of the dn, the name property check shouldn't be needed but can be replaced with 'descr'